src/witness.rs

```diff
@@ -218,6 +218,25 @@ impl<'de> Deserialize<'de> for WitnessName {
     }
 }
 
+impl Serialize for WitnessValues {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        use serde::ser::SerializeMap;
+
+        let mut map = serializer.serialize_map(Some(self.len()))?;
+        for (name, value) in self.iter() {
+            let value_map = BTreeMap::from([
+                ("value", value.to_string()),
+                ("type", value.ty().to_string()),
+            ]);
+            map.serialize_entry(name.as_inner(), &value_map)?;
+        }
+        map.end()
```

Comment:
In 5d7a91a4dd9b21a98e3b9f034c18ff9eb5a723d2:

Heh, there is definitely a more efficient way to do this than to actually construct a `BTreeMap` and serialize it. But I wouldn't worry about it for now. It's an opportunity for future optimization.